### DISCUSSION TITLE
Nuxt content integration?

### DISCUSSION THREAD

**Discussion by AlexLavoie42** [vriteio/vrite#47] [Ideas]
Full support for [Nuxt content MDC](https://github.com/nuxtlabs/vscode-mdc) would let you create a fully self hosted, git based cms with a proper editor using nuxt 3!

**Comment by areknawo** [comment]
@AlexLavoie42 I'm definitely looking at developing more integrations soon. Curious how you would see Vrite integrating with Nuxt Content. Especially given both are CMS or CMS-like.

One idea for MDC would be to add a transformer extension (like the one currently for MDX). Then you can use Vrite as a content editor for a Git repo, connected with Nuxt.

**Reply by AlexLavoie42** [reply]
That is essentially exactly what I was thinking. Basically I would like to recreate something like [Nuxt studio](https://nuxt.studio/) but have it be entirely self hosted.

**Reply by areknawo** [reply]
Vrite can already serve as a kind of headless CMS - both API and Git-based, e.g.:

- [Vrite Docs](https://docs.vrite.io) are edited in Vrite and [synced with the repo](https://github.com/vriteio/vrite/tree/main/apps/docs); Astro builds a static website from the MDX content, and Vrite API is used to add search.
- [Vrite Blog](https://vrite.io/blog) is also an Astro-powered static website, but [the content is fetched](https://github.com/vriteio/vrite/tree/main/apps/landing-page/src/pages/blog) straight from the Vrite API (in build phase) and is "transformed" from JSON (accessible via the API) to HTML in Astro.

Would love to somehow add a "Live preview" feature to Vrite in the future though, as it stands, it's not optimized for that.

TL;DR; Though not entirely similar, it's possible to use Vrite as a self-hosted Nuxt studio alternative.

**Reply by areknawo** [reply]
I thought about this more and "Live preview" should be doable for Vrite API/SDK-powered websites with only some optimization on the website side. Won't be as tightly integrated as Nuxt Studio, but will be framework-agnostic. No need for MDC. Will consider adding this for v1.
